src/coreComponents/physicsSolvers/wavePropagation/AcousticVTIWaveEquationSEM.cpp

```diff
@@ -0,0 +1,871 @@
+  NodeManager & nodeManager = domain.getMeshBody( 0 ).getMeshLevel( m_discretizationName ).getNodeManager();
+
+  arrayView1d< real32 > const p_nm1 = nodeManager.getField< fields::Pressure_p_nm1 >();
+  arrayView1d< real32 > const p_n = nodeManager.getField< fields::Pressure_p_n >();
+  arrayView1d< real32 > const p_np1 = nodeManager.getField< fields::Pressure_p_np1 >();
+
+  arrayView1d< real32 > const q_nm1 = nodeManager.getField< fields::Pressure_q_nm1 >();
+  arrayView1d< real32 > const q_n = nodeManager.getField< fields::Pressure_q_n >();
+  arrayView1d< real32 > const q_np1 = nodeManager.getField< fields::Pressure_q_np1 >();
+
+  ArrayOfArraysView< localIndex const > const faceToNodeMap = faceManager.nodeList().toViewConst();
+
+  /// array of indicators: 1 if a face is on on free surface; 0 otherwise
+  arrayView1d< localIndex > const freeSurfaceFaceIndicator = faceManager.getField< fields::FreeSurfaceFaceIndicator >();
+
+  /// array of indicators: 1 if a node is on on free surface; 0 otherwise
+  arrayView1d< localIndex > const freeSurfaceNodeIndicator = nodeManager.getField< fields::FreeSurfaceNodeIndicator >();
+
+  //freeSurfaceFaceIndicator.zero();
+  //freeSurfaceNodeIndicator.zero();
+
+  fsManager.apply< FaceManager >( time,
+                                  domain.getMeshBody( 0 ).getMeshLevel( m_discretizationName ),
+                                  string( "FreeSurface" ),
+                                  [&]( FieldSpecificationBase const & bc,
+                                       string const &,
+                                       SortedArrayView< localIndex const > const & targetSet,
+                                       FaceManager &,
+                                       string const & )
+  {
+    string const & functionName = bc.getFunctionName();
+
+    if( functionName.empty() || functionManager.getGroup< FunctionBase >( functionName ).isFunctionOfTime() == 2 )
+    {
+      real64 const value = bc.getScale();
+
+      for( localIndex i = 0; i < targetSet.size(); ++i )
+      {
+        localIndex const kf = targetSet[ i ];
+        freeSurfaceFaceIndicator[kf] = 1;
+
+        localIndex const numNodes = faceToNodeMap.sizeOfArray( kf );
+        for( localIndex a=0; a < numNodes; ++a )
+        {
+          localIndex const dof = faceToNodeMap( kf, a );
+          freeSurfaceNodeIndicator[dof] = 1;
+
+          p_np1[dof] = value;
+          p_n[dof]   = value;
+          p_nm1[dof] = value;
+
+          q_np1[dof] = value;
+          q_n[dof]   = value;
+          q_nm1[dof] = value;
+
+        }
+      }
+    }
+    else
+    {
+      GEOS_ERROR( "This option is not supported yet" );
+    }
+  } );
+}
+
+
+/**
+ * Checks if a directory exists.
+ *
+ * @param dirName Directory name to check existence of.
+ * @return true is dirName exists and is a directory.
+ */
+bool AcousticVTIWaveEquationSEM::dirExists( const std::string & dirName )
```

Comment:
This should be moved but I think it can wait the refactoring that @acitrain is preparing

Reply:
Yes totally. I talked to you about this function. It exists in `AcousticWaveEquationSEM` but globally (not within a namespace). I "had to" copy it too...